build-ps/percona-server-5.7_builder.sh

```diff
@@ -0,0 +1,825 @@
+    dpkg-buildpackage -S
+
+    cd ${WORKDIR}
+
+    mkdir -p $WORKDIR/source_deb
+    mkdir -p $CURDIR/source_deb
+    cp *.debian.tar.* $WORKDIR/source_deb
+    cp *_source.changes $WORKDIR/source_deb
+    cp *.dsc $WORKDIR/source_deb
+    cp *.orig.tar.gz $WORKDIR/source_deb
+    cp *.debian.tar.* $CURDIR/source_deb
+    cp *_source.changes $CURDIR/source_deb
+    cp *.dsc $CURDIR/source_deb
+    cp *.orig.tar.gz $CURDIR/source_deb
+}
+
+build_deb(){
+    if [ $DEB = 0 ]
+    then
+        echo "source deb package will not be created"
+        return;
+    fi
+    if [ "x$OS" = "xrpm" ]
+    then
+        echo "It is not possible to build source deb here"
+        exit 1
+    fi
+    for file in 'dsc' 'orig.tar.gz' 'changes' 'debian.tar*'
+    do
+        get_deb_sources $file
+    done
+    cd $WORKDIR
+    rm -fv *.deb
+
+    export DEBIAN_VERSION="$(lsb_release -sc)"
+
+    DSC=$(basename $(find . -name '*.dsc' | sort | tail -n 1))
+    DIRNAME=$(echo ${DSC%-${DEB_RELEASE}.dsc} | sed -e 's:_:-:g')
+    VERSION=$(echo ${DSC} | sed -e 's:_:-:g' | awk -F'-' '{print $4}')
+    RELEASE=$(echo ${DSC} | sed -e 's:_:-:g' | awk -F'-' '{print $5}')
+    ARCH=$(uname -m)
+    export EXTRAVER=${MYSQL_VERSION_EXTRA#-}
+    #
+    echo "ARCH=${ARCH}" >> percona-server-5.7.properties
+    echo "DEBIAN_VERSION=${DEBIAN_VERSION}" >> percona-server-5.7.properties
+    echo "VERSION=${VERSION}" >> percona-server-5.7.properties
+    #
+
+    dpkg-source -x ${DSC}
+
+    cd ${DIRNAME}
+    #
+    if [ ${DEBIAN_VERSION} = xenial -o ${DEBIAN_VERSION} = artful -o ${DEBIAN_VERSION} = bionic -o ${DEBIAN_VERSION} = trusty ]; then
+        rm -rf debian
+        cp -r build-ps/ubuntu debian
+    fi
+    dch -b -m -D "$DEBIAN_VERSION" --force-distribution -v "${VERSION}-${RELEASE}-${DEB_RELEASE}.${DEBIAN_VERSION}" 'Update distribution'
+
+    if [ ${ARCH} != "x86_64" ]; then
+        rm -f debian/rules
+        rm -f debian/control
+        mv debian/rules.notokudb debian/rules
+        mv debian/control.notokudb debian/control
+    else
+        if [ ${DEBIAN_VERSION} != trusty -a ${DEBIAN_VERSION} != xenial -a ${DEBIAN_VERSION} != jessie -a ${DEBIAN_VERSION} != stretch -a ${DEBIAN_VERSION} != artful -a ${DEBIAN_VERSION} != bionic ]; then
+            gcc47=$(which gcc-4.7 2>/dev/null || true)
+            if [ -x "${gcc47}" ]; then
+                export CC=gcc-4.7
+                export USE_THIS_GCC_VERSION="-4.7"
+                export CXX=g++-4.7
+            else
+                export CC=gcc-4.8
+                export USE_THIS_GCC_VERSION="-4.8"
+                export CXX=g++-4.8
+            fi
+        fi
+    fi
+
+    if [ ${DEBIAN_VERSION} = "xenial" ]; then
+        sed -i 's/export CFLAGS=/export CFLAGS=-Wno-error=date-time /' debian/rules 
+        sed -i 's/export CXXFLAGS=/export CXXFLAGS=-Wno-error=date-time /' debian/rules
+    fi
+
+    if [ ${DEBIAN_VERSION} = "stretch" ]; then
```

Comment:
custom cflags like -Wno-* to disable errors are just workarounds to build the code of uneven quality. release code should be ready and it shouldn't require any adjustments. Roel and Laurynas can say more or make a corrections